syntax/std.go

```diff
@@ -51,8 +51,19 @@ func StdScope() rel.Scope {
 					case rel.Dict:
 						attrs := make([]rel.Attr, 0, t.Count())
 						for e := t.DictEnumerator(); e.MoveNext(); {
-							key, value := e.Current()
-							attrs = append(attrs, rel.NewAttr(key.(rel.String).String(), value))
+							keyVal, value := e.Current()
+							key := ""
+							if keyVal.IsTrue() {
+								key = keyVal.String()
+							} else {
+								switch keyVal.(type) {
+								case rel.Number:
+									key = "0"
+								case rel.Tuple:
+									key = "()"
+								}
```

Comment:
Here's a more detailed explanation following up from my previous comment. Requiring programmers to be explicit about their intent is an important element of arr.ai's design. In particular, the language should avoid automatically stringifying for convenience. Doing so opens a can of worms. Consider `//tuple({(): 1, '()': 2})` and `//tuple({"abc": 3, <<"abc">>, 4})`. Also consider that `//tuple({[1, 2]: 42, '[1,2]': 43})` would work fine, but `//tuple({[1, 2]: 42, '[1, 2]': 43})` would croak, leading to much head-scratching. Similarly for `//tuple([{true, false}: 42, '{true, false}': 43}`.

In arr.ai, it is very easy to transform keys to strings when desired, which also gives the author greater awareness of what they're getting themselves into and more control over how it's done, e.g.:

```arrai
let jsonPathValues = {
    ['a', 'b', 0]: 42,
    ['a', 'b', 1]: 43,
    ['c', 'd', 2]: 43,
};
//tuple(jsonPathValues => (@: $"${.@::_}", :.@value))
```

Reply:
Makes sense, thanks for the explanation.